apps/gitness/src/framework/routing/types.ts

```diff
@@ -99,6 +99,11 @@ export interface CustomHandle {
    * Associated route name for the breadcrumb.
    */
   routeName?: string
+
+  /**
+   * Updates the document title based on route parameters.
+   */
+  documentTitle?: (params: Params<string>) => string
```

Comment:
can we support an API like this?

```suggestion
  documentTitle?: string | (params: Params<string>) => string
```

this would allow the user to pass a hardcoded string where params are not needed.

Reply:
Done.